src/lib/avatar.ts

```diff
@@ -65,10 +65,21 @@ export async function uploadUserAvatar(avatar: File) {
 export const getUserAvatar = async (txId: string) => {
   try {
     const data = await arweave.transactions.getData(txId, { decode: true });
-    const blob = new Blob([data], { type: "image" });
-    const imageUrl = URL.createObjectURL(blob);
+    let mimeType = "image/png";
+    if (data instanceof Uint8Array) {
```

Comment:
I'm getting instances of `ArrayBuffer` instead of `Uint8Array` here for svgs. We could test for both, wdyt?

Reply:
@7i7o added 
`    const buffer = data instanceof ArrayBuffer ? new Uint8Array(data) : data;

    if (buffer instanceof Uint8Array) {
      const textDecoder = new TextDecoder("utf-8");
      const snippet = textDecoder.decode(buffer.slice(0, 80));

      if (snippet.includes("<svg")) {
        mimeType = "image/svg+xml";
      }
    }`